File: app/src/main/java/site/aiok/onepic/model/PuzzlePiece.kt
package site.aiok.onepic.model

import android.graphics.Bitmap

data class PuzzlePiece(
    val id: Int,
    var currentX: Float,
    var currentY: Float,
    var targetX: Float,
    var targetY: Float,
    var width: Float,
    var height: Float,
    var bitmap: Bitmap,
    var zIndex: Int,
    var isLocked: Boolean = false,
    var groupId: Int = -1,
    var row: Int,  // 改为 var 以支持网格位置更新
    var col: Int,  // 改为 var 以支持网格位置更新
)

File: ios/OnePic/OnePic/Model/PuzzlePiece.swift
import Foundation
import SwiftUI
import UIKit

// Verified: PuzzlePiece is a struct.
// In GameViewModel loop `for i in 0..<self.pieces.count`, we are modifying `self.pieces[i]`.
// Since `pieces` is @Published var, this is correct and will trigger UI update.

struct PuzzlePiece: Identifiable {
    enum Edge {
        case top, bottom, left, right
    }
    
    let id: Int
    var currentX: CGFloat
    var currentY: CGFloat
    var targetX: CGFloat
    var targetY: CGFloat
    var width: CGFloat
    var height: CGFloat
    var image: UIImage
    var zIndex: Int
    var isLocked: Bool = false
    var groupId: Int = -1
    var row: Int
    var col: Int
    var hiddenEdges: Set<Edge> = []
}
